lxd/api_cluster.go

```diff
@@ -4599,3 +4618,675 @@ func autoHealCluster(_ context.Context, s *state.State, offlineMembers []db.Node
 
 	return nil
 }
+
+// swagger:operation GET /1.0/clusters cluster-links cluster_links_get
+//
+//	Get cluster links
+//
+//	Gets cluster links.
+//
+//	---
+//	produces:
+//	  - application/json
+//	responses:
+//	  "200":
+//	    description: Cluster links
+//	    schema:
+//	      type: object
+//	      description: Sync response
+//	      properties:
+//	        type:
+//	          type: string
+//	          description: Response type
+//	          example: sync
+//	        status:
+//	          type: string
+//	          description: Status description
+//	          example: Success
+//	        status_code:
+//	          type: integer
+//	          description: Status code
+//	          example: 200
+//	        metadata:
+//	          $ref: "#/definitions/ClusterLink"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func clusterLinksGet(d *Daemon, r *http.Request) response.Response {
+	s := d.State()
+
+	var resp []*api.ClusterLink
+
+	err := s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		clusterLinks, err := dbCluster.GetClusterLinks(ctx, tx.Tx())
+		if err != nil {
+			return fmt.Errorf("failed to fetch cluster links: %w", err)
+		}
+
+		for _, clusterLink := range clusterLinks {
+			clusterLinkAPI, err := clusterLink.ToAPI(ctx, tx.Tx())
+			if err != nil {
+				return err
+			}
+
+			resp = append(resp, clusterLinkAPI)
+		}
+
+		return nil
+	})
+	if err != nil {
```

Comment:
Also consider the `with-entitlements` query parameter, for enriching responses with requested entitlements.